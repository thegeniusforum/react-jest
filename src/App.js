import React, { Component } from "react";
import Account from "./Account";

class App extends Component {
  state = {
    isLoading: true,
    users: [],
    error: null,
  };

  fetchUsers() {
    fetch(`https://jsonplaceholder.typicode.com/users`)
      .then((response) => response.json())
      .then((data) =>
        this.setState({
          users: data,
          isLoading: false,
        })
      )
      .catch((error) => this.setState({ error, isLoading: false }));
  }

  componentDidMount() {
    this.fetchUsers();
  }

  render() {
    const { isLoading, users, error } = this.state;
    return (
      <React.Fragment>
        <h1>Display Active Users Account Details</h1>
        {error ? <p>{error.message}</p> : null}
        {!isLoading ? (
          users.map((user) => {
            return <Account key={user.username} user={user} />;
          })
        ) : (
          <h3>Loading Users...</h3>
        )}
      </React.Fragment>
    );
  }
}

export default App;
