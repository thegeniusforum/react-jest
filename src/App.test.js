import { shallow, mount } from "enzyme";
import App from "./App";
import React from "react";
import ReactDOM from "react-dom";
import Account from "./Account";
import toJson from "enzyme-to-json";

it("Renders App without crashing", () => {
  shallow(<App />);
});

it("Renders Account Header", () => {
  const wrapper = shallow(<App />);
  const message = <h1>Display Active Users Account Details</h1>;

  expect(wrapper.contains(message)).toEqual(true);
});

it.skip("Renders without crashing", () => {
  shallow(<App />);
});

const user = {
  name: "Adeneye David",
  email: "david@gmail.com",
  username: "Dave",
};

describe("Testing Props", () => {
  it("Accepts user props", () => {
    const wrapper = mount(<Account user={user} />);
    expect(wrapper.props().user).toEqual(user);
  });

  it("contains users account email", () => {
    const wrapper = mount(<Account user={user} />);
    const value = wrapper.find("p").text();
    expect(value).toEqual(user.email);
  });
});

it("Renders without errors", () => {
  const wrapper = mount(<App />);
  expect(wrapper.state("error")).toEqual(null);
});

it("Snapshot of tree renders correctly", () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});
