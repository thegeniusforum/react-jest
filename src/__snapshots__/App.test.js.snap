// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of tree renders correctly 1`] = `
<Fragment>
  <h1>
    Display Active Users Account Details
  </h1>
  <h3>
    Loading Users...
  </h3>
</Fragment>
`;
